spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/RNN.scala

```diff
@@ -257,9 +297,9 @@ class RNN(
         dist_iter_MD = dst_iter.getMemoryDescription()
 
         src_i = new TensorMMap(inputShape_iter)
-        src_i.dense.copy(Tensor[Float]().resize(inputShape_iter).zero())
         dst_i = new TensorMMap(outputShape_iter)
-        dst_i.dense.copy(Tensor[Float]().resize(outputShape_iter).zero())
+        gradsrc_i = new TensorMMap(inputShape_iter)
```

Comment:
for those three case, there are some duplicate code, please do code refactor.

Reply:
Changed. Combined into one using case condition.